src/plugins/expressions/public/react_expression_renderer/use_expression_renderer.ts

```diff
@@ -77,6 +79,10 @@ export function useExpressionRenderer(
   // will call done() in LayoutEffect when done with rendering custom error state
   const errorRenderHandlerRef = useRef<IInterpreterRenderHandlers | null>(null);
 
+  if (abortController?.signal)
+    abortController.signal.onabort = () => {
+      expressionLoaderRef.current?.cancel();
+    };
```

Comment:
Would it be better to add this as an event listener (within an effect to avoid duplicate registrations) instead of setting the property to avoid accidentally overwriting a callback that might have already been set?